### PR TITLE
hacktoberfest: migrate 'docker custom build environment' wiki to github

### PR DESCRIPTION
**Pull request by kaltepeter**
Fixes: https://github.com/jenkins-infra/jenkins.io/issues/3789

Depends on: https://github.com/jenkinsci/docker-custom-build-environment-plugin/pull/75

The existing link: `RewriteRule "^/display/JENKINS/CloudBees\+Docker\+Custom\+Build\+Environment\+Plugin$" "https://plugins.jenkins.io/docker-custom-build-environment" [NC,L,QSA,R=301]` appears to have been just one of two redirect links or incorrect.

image

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
@kaltepeter did you want this reviewed?  If so, then you need to press the "Ready for review" button.  I confirmed that there is a wiki page at the URL this pull request is redirecting.  This seems like the right thing to do.

**Comment by kaltepeter** [comment]
@MarkEWaite I believe this is ready too. I wasn't sure if https://github.com/jenkinsci/docker-custom-build-environment-plugin/pull/75 needed to be merged and released first.

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
> @MarkEWaite I believe this is ready too. I wasn't sure if [jenkinsci/docker-custom-build-environment-plugin#75](https://github.com/jenkinsci/docker-custom-build-environment-plugin/pull/75) needed to be merged and released first.

I believe that the `RewriteCond` line that follows your addition will allow the wiki exporter to continue reading the documentation from the wiki page while the line you added will redirect users to the plugins site rather than having them read the pages from the wiki.

However, I'm weak on my comprehension of `RewriteCond` and when it is required and when it is not required.

@halkeye or @olblak do you have more insights to share on this case?

* Plugin docs are on the [wiki](https://wiki.jenkins.io/display/JENKINS/Docker+Custom+Build+Environment+Plugin), not in [GitHub](https://github.com/jenkinsci/docker-custom-build-environment-plugin/blob/master/README.md)
* Would like to redirect users from wiki to [plugins.jenkins.io](https://plugins.jenkins.io/docker-custom-build-environment/) without redirecting the plugin site generator

Given those two conditions, I think that we need the RewriteRule that @kaltepeter has proposed and we need the `RewriteCond` that is already in the file.

**Comment by halkeye** [comment]
Yea, the rewite condition says rewrite this url if the user agent is not(the !) ^jenkins-wiki-exporter/(.*)$

File: dist/profile/templates/confluence/vhost.conf [hunk 1]
```diff
@@ -837,6 +837,7 @@ RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
 RewriteRule "^/display/JENKINS/Docker\+Compose\+Build\+Step\+Plugin$" "https://plugins.jenkins.io/docker-compose-build-step" [NC,L,QSA,R=301]
 RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
 RewriteRule "^/display/JENKINS/CloudBees\+Docker\+Custom\+Build\+Environment\+Plugin$" "https://plugins.jenkins.io/docker-custom-build-environment" [NC,L,QSA,R=301]
+RewriteRule "^/display/JENKINS/Docker\+Custom\+Build\+Environment\+Plugin$" "https://plugins.jenkins.io/docker-custom-build-environment" [NC,L,QSA,R=301]
```
Comment (halkeye) [inline dist/profile/templates/confluence/vhost.conf:435]:
```suggestion
RewriteRule "^/display/JENKINS/Docker\+Custom\+Build\+Environment\+Plugin$" "https://plugins.jenkins.io/docker-custom-build-environment" [NC,L,QSA,R=301]
RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
```

**Pull request merged**